### PR TITLE
management: users must acknowledge use of beta software

### PR DESCRIPTION
**Pull request by tbielawa**
Adds a new role variable, openshift_management_install_beta. This
variable defaults to false. The value of this variable is checked
during the validation phase.

* If true, the install will not continue.

* If false, The user is presented with an informative message letting
  them know this is beta software and there is low/no support at this
  time. The installation will abort and instruct the user how to
  continue.

### CODE REVIEW

File: roles/openshift_management/tasks/validate.yml [hunk 1]
```diff
@@ -2,12 +2,23 @@
 # Validate configuration parameters passed to the openshift_management role
 
 ######################################################################
+# BETA ACKNOWLEDGEMENT
+- name: Ensure BETA software notice has been acknowledged
+  assert:
+    that:
+      - openshift_management_install_beta | default(false) | bool
```
Comment (sdodson) [inline roles/openshift_management/tasks/validate.yml]: we said we'd only do this for OCP, `or openshift_deployment_type != 'openshift-enterprise'`

**Comment by tbielawa** [comment]
bug. `openshift_deployment_type` isn't enough by itself.

**Comment by tbielawa** [comment]
@sdodson PTAL again. The old 'when' condition wasn't covering the duplicate 'deployment_type' vs. 'openshift_deployment_type' problem.

**Comment by sdodson** [comment]
> @sdodson PTAL again. The old 'when' condition wasn't covering the duplicate 'deployment_type' vs. 'openshift_deployment_type' problem.

you shouldn't have to worry about that if std_include had been called.

**Comment by tbielawa** [comment]
@sdodson I guess it's not getting ran then. From the product installer playbook `playbooks/byo/openshift-management/config.yml` these are the only ones included

```
---
- include: ../openshift-cluster/initialize_groups.yml

- include: ../../common/openshift-cluster/evaluate_groups.yml

- include: ../../common/openshift-management/config.yml
```

So I added a call to `openshift_facts` in the `validate` management playbook and went to `openshift.common.deployment_type` instead.

**Pull request merged**